### ISSUE TITLE
Inline admonitions/details overlap other details

### ISSUE THREAD

**Issue by danierutu** [squidfunk/mkdocs-material#2336]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

When you set an admonition or detail block as inline, following admonition blocks will adapt to its size, but detail blocks will stay unaltered. Hence, the detail block ignores the inline block and gets overlapped.

I have tried both `inline` and `inline end`, also both an inline admonition and an inline detail. The result is the same, as you can see in the screenshots below.

<details>
<summary><b>See screenshots</b></summary>

![image1](https://user-images.githubusercontent.com/63101447/108916114-1b170a00-762e-11eb-8ab9-1f9ac720dbe6.png)
![image2](https://user-images.githubusercontent.com/63101447/108916674-e192ce80-762e-11eb-96f7-003f5e546e8a.png)
![image3](https://user-images.githubusercontent.com/63101447/108916812-14d55d80-762f-11eb-8f92-788e6f66044c.png)

</details>

### Steps to reproduce the bug

<details>
<summary><b>Use a setup similar to this</b></summary>

```
??? info inline "I'm inline!"
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer commodo libero a massa consectetur sollicitudin.
    Morbi molestie nisl tempor turpis eleifend tristique. Donec ac ante interdum, lacinia ex eget, porttitor nisi.
    Pellentesque nec massa auctor felis sagittis cursus id et nisl. Duis aliquam, felis ut porttitor sollicitudin,
    libero libero placerat dui, laoreet tristique nibh nibh vel nunc. Quisque nec volutpat arcu. Pellentesque sit amet
    nibh porttitor, feugiat ipsum ut, commodo nisl.

!!! info "Admonition 1"
    I am an admonition

??? info "Detail 1"
    I am a detail

!!! info "Admonition 2"
    I am another admonition

```
</details>

### Package versions

* Python: 3.9.1
* MkDocs: 1.1.2
* Material: 7.0.0

### System information

* OS: Windows 21H1
* Browser: Firefox 85

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting!

**Comment by squidfunk** [comment]
Fixed in a17b6375 together with #2337.

**Comment by squidfunk** [comment]
Released as part of 7.0.1.
